include/bluetooth/audio/cap.h

```diff
@@ -86,6 +137,210 @@ struct bt_cap_cb {
  */
 int bt_cap_find_ccid_service_service(struct bt_conn *conn, uint8_t ccid);
 
+/**
+ * @brief Discovers audio support on a remote device.
+ *
+ * This will discover the Common Audio Service (CAS) on the remote device, to
+ * if the remote device supports the Common Audio Profile.
+ *
+ * @param conn Connection to a remote server.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_cap_unicast_discover(struct bt_conn *conn);
+
+/** Type of CAP set */
+enum bt_cap_set_type {
+	/** The set is an ad-hoc set */
+	BT_CAP_SET_TYPE_AD_HOC,
+	/** The set is a CSIP Coordinated Set */
+	BT_CAP_SET_TYPE_CSIP,
+};
+
+struct bt_cap_unicast_audio_start_param {
+	/** The type of the set */
+	enum bt_cap_set_type type;
+
+	/** The number of set members in @p members and number of streams in @p streams */
+	uint8_t count;
+
+	union {
+		/**
+		 * @brief List of connection pointers if @p type is BT_CAP_SET_TYPE_AD_HOC
+		 *
+		 * The number of pointers in this array shall be at least @p count
+		 */
+		struct bt_conn **members;
+
+		/** CSIP Coordinated Set struct used if @p type is BT_CAP_SET_TYPE_CSIP */
+		struct {
+			/**
+			 * @brief CSIP Set Members in the set
+			 *
+			 * The number of pointers in this array shall be at least %p count
+			 */
+			struct bt_csis_client_set_member **members;
+
+			/** Set information used to verify the set @p members */
+			struct bt_csis_client_set_info *set_info;
+		} csip;
+	};
+
+	/** @brief Streams for the @p members
+	 *
+	 * stream[i] will be associated with members[i] if not already
+	 * initialized, else the stream will be verified against the member.
+	 */
+	struct bt_audio_stream **streams;
+
+	/**
+	 * @brief Codec configuration
+	 *
+	 * The @p codec.meta shall include a list of CCIDs as well as a non-0
+	 * context bitfield.
+	 */
+	const struct bt_codec *codec;
+
+	/** Quality of Service configuration */
+	const struct bt_codec_qos *qos;
```

Comment:
Maybe instead of bt_codec_qos we could provide content type and target latency which later could be used by CAP to choose preferred QoS?

If we want to keep it this way I would call it target_qos, because we might end up with different QoS parameters based on the remote preference.

Reply:
> Maybe instead of bt_codec_qos we could provide content type and target latency which later could be used by CAP to choose preferred QoS?

I originally designed it like that, but there are so many different values that go into the metadata and codec config that having an API that allows to set all these is near impossible, unless we split up the API to a `start_prepare` and a `start_commit` type of deal. It is much more flexible and simpler for the API to simply request the entire `bt_codec_qos` structure, and then verify the content (i.e. number of channels, etc.). 

> If we want to keep it this way I would call it target_qos, because we might end up with different QoS parameters based on the remote preference.

How? The remote preference is only a preference, and if the remote device actively rejects our QoS, then this procedure will fail. I don't think we should modify the requested QoS from the application autonomously.

Reply:
> How? The remote preference is only a preference, and if the remote device actively rejects our QoS, then this procedure will fail. I don't think we should modify the requested QoS from the application autonomously.

If we would not take audio server preference into account, it is not Nice but some Initiators can do it. But e.g. we need to take into account min/max presentation delay.

Reply:
I'm not fully opposed to changing this to something that is a preference from the application, rather than a specific requirement. In that case, perhaps `bt_codec_qos_pref` could/should be used instead, but then we are exposing an API where the user cannot fully control the values, and I am not sure if that is a better or worse API that getting an error returned if the server rejects the request. 

@asbjornsabo Do you have any opinion?